azure/services/aso/aso.go

```diff
@@ -200,6 +201,9 @@ func (r *reconciler[T]) CreateOrUpdateResource(ctx context.Context, spec azure.A
 
 	diff := cmp.Diff(existing, parameters)
 	if diff == "" {
+		if readyErr != nil {
+			return zero, readyErr
```

Comment:
I am curious, why are we trying to return `zero-valued T` instead of `parameters` if the `diff == ""`
_Diff_ would be empty if the _desired spec matches existing spec_, shouldn't we return the spec instead of the `zero T`?

Reply:
This is following what I think is a fairly standard pattern to only return either a non-zero value or a non-nil error, and not a non-zero value with a non-nil error. That's confusing to me when I see that. "Is the value safe to use?" "Did this really fail?"

If we find a valid use case for doing that anyway though I don't think it would be a huge deal to change.

Reply:
Makes sense. 

A short comment explaining this change would greatly help anyone visiting this code. The point to be noted here is that "Parameters" func gets called at least once, updating the ASO resource and CAPZ waits until the changes get reconciled before sending another request.